skimage/morphology/rolling_ball.py

```diff
@@ -0,0 +1,197 @@
+        than 0.
+    has_nan: bool, optional
+        If ``False`` (default) assumes that none of the values in ``image``
+        are ``np.nan``, and uses a faster implementation.
+    num_threads: int, optional
+        The maximum number of threads to use. If ``None`` use the OpenMP
+        default value; typically equal to the maximum number of virtual cores.
+        Note: This is an upper limit to the number of threads. The exact number
+        is determined by the system's OpenMP library.
+
+    Returns
+    -------
+    background : ndarray
+        The estimated background of the image.
+
+    Notes
+    -----
+    - For the pixel that has its background intensity estimated (without loss
+      of generality at ``(0,0)``) the rolling ellipsoid method places an
+      ellipsoid under it and raises the ellipsoid until its surface touches the
+      image umbra at ``pos=(y,x)``. The background intensity is then estimated
+      using the image intensity at that position (``image[y, x]``) plus the
+      difference of ``intensity_vertex`` and the surface of the ellipsoid at
+      ``pos``. The surface intensity of the ellipsoid is computed using the
+      canonical ellipsis equation
+      .. code-block:: python
+
+        | semi_spatial = kernel_shape / 2
+        | semi_vertex = intensity_vertex / 2
+        | np.sum((pos/semi_spatial)**2) + (intensity/semi_vertex)**2 = 1
+
+    - This algorithm assumes that dark pixels correspond to the background. If
+      you have a bright background, invert the image before passing it to the
+      function, e.g., using `utils.invert`.
+    - This algorithm is sensitive to noise (in particular salt-and-pepper
+      noise). If this is a problem in your image, you can apply mild
+      gaussian smoothing before passing the image to this function.
+
+    References
+    ----------
+    .. [1] Sternberg, Stanley R. "Biomedical image processing." Computer 1
+           (1983): 22-34. :DOI:`10.1109/MC.1983.1654163`
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skimage import data
+    >>> from skimage.morphology import rolling_ellipsoid
+    >>> image = data.coins()
+    >>> background = rolling_ellipsoid(data.coins())
+    >>> filtered_image = image - background
+    """
+
+    if num_threads is None:
+        num_threads = 0
+
+    kernel_shape = np.asarray(kernel_shape)
+    if kernel_shape.ndim == 0:
+        kernel_shape = np.full_like(image.shape, kernel_shape)
+
+    kernel_shape_int = np.asarray(kernel_shape // 2 * 2 + 1, dtype=np.intp)
+
+    intensity_vertex = np.asarray(intensity_vertex, dtype=np.float_)
+    intensity_vertex = intensity_vertex / 2
+
+    image = np.asarray(image)
+    img = image.astype(np.float_)
+
+    ellipsoid_coords = np.stack(
+        np.meshgrid(
+            *[range(-x, x + 1) for x in kernel_shape_int // 2],
+            indexing='ij'
+        ),
+        axis=-1).reshape(-1, len(kernel_shape))
+    tmp = np.sum(
```

Comment:
What is tmp? Can we come up with a more descriptive name?

Reply:
tmp is a temporary variable x). Its sole purpose is to allow me to comply with PEP8 and not have the code look super horrible with weird line breaks and indentations (it's already borderline).

Suggestions for a better name are welcome :)